components/Survey.js

```diff
@@ -0,0 +1,55 @@
+const React = require('react');
+const Rating = require('./Rating');
+
+function random(array) {
+  return array[Math.floor(Math.random() * array.length)];
+}
+
+class Survey extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      rating: 0
+    };
+
+    this.increaseRating = this.increaseRating.bind(this);
+    this.maintainRating = this.maintainRating.bind(this);
+    this.decreaseRating = this.decreaseRating.bind(this);
+  }
+
+  increaseRating() {
+    this.setState({ rating: this.state.rating + 1 });
+  }
+
+  decreaseRating() {
+    this.setState({ rating: this.state.rating - 1 });
+  }
+
+  maintainRating() {
```

Comment:
I think we do need it, because it causes the `Rating` component to re-render and thus the `trend = 'stable'` will be displayed. (in `componentWillReceiveProps` for `Rating` we compare current and new props etc)